addons/mail/static/src/core/common/thread_model.js

```diff
@@ -917,7 +917,7 @@ export class Thread extends Record {
                 }
             });
         }
-        if (this.needactionMessages.length > 0) {
+        if (this.needactionCounter > 0) {
```

Comment:
This is not 100% correct IMO as `needactionCounter` is a getter that can either return `message_unread_counter` or `message_needaction_counter`. In the first case, it would call `mark_all_as_read` while it is not needed. I think we should directly check `message_needaction_counter`.

Reply:
You  are right,